examples/pipelines/simple_pipeline/components/load_from_hub/src/main.py

```diff
@@ -0,0 +1,74 @@
+"""
+This component loads a seed dataset from the hub.
+"""
+import io
+import logging
+
+from datasets import load_dataset
+
+import dask.dataframe as dd
+import dask.array as da
+import pandas as pd
+import numpy as np
+from PIL import Image
+
+from express.dataset import FondantComponent
+from express.logger import configure_logging
+
+configure_logging()
+logger = logging.getLogger(__name__)
+
+
+def extract_width(image_bytes):
+    # Decode image bytes to PIL Image object
+    pil_image = Image.open(io.BytesIO(image_bytes["bytes"]))
+    width = pil_image.size[0]
+    
+    return width
+
+def extract_height(image_bytes):
+    # Decode image bytes to PIL Image object
+    pil_image = Image.open(io.BytesIO(image_bytes["bytes"]))
+    height = pil_image.size[1]
+    
+    return height
+
+
+class LoadFromHubComponent(FondantComponent):
+    type = "load"
+
+    @classmethod
+    def load(cls, args):
+        """
+        Args:
+            args: additional arguments passed to the component
+        
+        Returns:
+            Dataset: HF dataset
+        """
+        # 1) Load data, read as Dask dataframe
+        logger.info("Loading dataset from the hub...")
+        dataset = load_dataset(args.dataset_name, split="train")
+        dataset.to_parquet("data.parquet")
+        dask_df = dd.read_parquet("data.parquet")
+
+        # 2) Add index to the dataframe
+        logger.info("Creating index...")
+        index_list = [idx for idx in range(len(dask_df))]
+        source_list = ["hub" for _ in range(len(dask_df))]
+
+        dask_df["id"] = da.array(index_list)
+        dask_df["source"] = da.array(source_list)
+
+        # 3) Rename columns
+        dask_df = dask_df.rename(columns={"image": "images_data", "text": "captions_data"})  
+        
+        # 4) Add width and height columns
+        dask_df['images_width'] = dask_df['images_data'].map(extract_width, meta=pd.Series([], dtype=np.float64))
+        dask_df['images_height'] = dask_df['images_data'].map(extract_height, meta=pd.Series([], dtype=np.float64))
```

Comment:
the `np.float64` does not really match the type specified in the subsets. We need a way to check for that or propagate them through, they seem a bit redundant for now

Reply:
Dask has some weird functionality, like why do I need to pass this `meta` thing with an empty Series?

Reply:
I noticed that as well. I read a bit on it and I think when you apply a function to a Dask DataFrame using the map() method, Dask needs to know the data type of the output to determine the optimal execution plan and to allocate the necessary resources for the computation.

Will have to look more into it